pkg/avancement/service_manager_test.go

```diff
@@ -75,3 +77,32 @@ func mustAddCredentials(t *testing.T, repoURL string, a *git.Author) string {
 	}
 	return u
 }
+
+func TestRepositoryCloneErrorOmitsToken(t *testing.T) {
+	dstBranch := ""
+	author := &git.Author{Name: "Testing User", Email: "testing@example.com", Token: "test-token"}
+	client, _ := fakescm.NewDefault()
+	fakeClientFactory := func(s string) *scm.Client {
+		return client
+	}
+	devRepo, stagingRepo := mock.New("/dev", "master"), mock.New("/staging", "master")
+	repos := map[string]*mock.Repository{
+		mustAddCredentials(t, dev, author):     devRepo,
+		mustAddCredentials(t, staging, author): stagingRepo,
+	}
+	sm := New("tmp", author)
+	sm.clientFactory = fakeClientFactory
+	sm.repoFactory = func(url, _ string, _ bool) (git.Repo, error) {
+		// This actually causes the error and results in trying to create a repository
+		// which can surface the token
+		return git.Repo(repos[url]), errors.New("status code 128")
+	}
+	devRepo.AddFiles("/services/my-service/base/config/myfile.yaml")
+
+	err := sm.Promote("my-service", dev, staging, dstBranch)
+	if err != nil {
+		if strings.Contains(err.Error(), "test-token") {
+			t.Fatalf("Error contained the access token, full error: %s", err.Error())
```

Comment:
Hmm, so with that method we could check the error message is a certain exact pattern (e.g. `failed to clone repository foo` instead of `failed to clone repository token:foo`) - that does tightly couple this test to our message format though whereas if it's this way we just keep it specific to not having the token - what do you think?

Reply:
I'd change this test, we know it should return an error, so...

```go
if err == nil {
    t.Fatal("did not get an error when processing a bad branch")
}
```

AssertMatch actually uses regular expressions to match, so, you can do things like `"error doing the thing.*https://example.com"`

Errors in tests tend to begin with a lower-case "s/Error/error/".